aws/structure.go

```diff
@@ -3059,3 +3059,23 @@ func flattenMqBrokerInstances(instances []*mq.BrokerInstance) []interface{} {
 
 	return l
 }
+
+func flattenApiGatewayStageCanarySettings(canarySettings *apigateway.CanarySettings) []interface{} {
+	settings := make(map[string]interface{})
+	overrides := make(map[string]string)
+
+	if canarySettings == nil {
+		return nil
+	}
+
+	for k, v := range canarySettings.StageVariableOverrides {
```

Comment:
The AWS SDK provides a convenience method which can replace this logic: `aws.StringValueMap()`